### PR TITLE
Added rules to build syncd and orchagent containers for Cavium target

### PR DESCRIPTION
**Pull request by garyachy**

### CODE REVIEW

File: platform/cavium/docker-syncd-cavm/Dockerfile.template [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM docker-base
+
+RUN apt-get update
+
+COPY deps /deps
+
+RUN apt-get -y install libpcap-dev libxml2-dev python-dev swig libsensors4-dev redis-tools
```
Comment (marian-pritsak) [inline platform/cavium/docker-syncd-cavm/Dockerfile.template]: redis-tools is installed from sources, not needed here
Why are dev packages needed?

Comment (garyachy) [inline platform/cavium/docker-syncd-cavm/Dockerfile.template]: 1. We need redis-tools in syncd container to synchronize with redis DB
2. We need dev packages for diagnostics shell support

Comment (marian-pritsak) [inline platform/cavium/docker-syncd-cavm/Dockerfile.template]: I don't see that other vendors install redis-tools into docker-syncd
Anyway, if you need it, redis is built from sources, so you should install it like [here](https://github.com/Azure/sonic-buildimage/blob/master/rules/docker-database.mk#L5)

Comment (lguohan) [inline platform/cavium/docker-syncd-cavm/Dockerfile.template]: yes, we are using redis 3.2.4 instead 3.0.7 in the debian jessie.

Comment (garyachy) [inline platform/cavium/docker-syncd-cavm/Dockerfile.template]: Now source based redis-tools are used

File: platform/cavium/rules.mk [hunk 2]
```diff
@@ -1,21 +1,13 @@
...
-$(XP_TOOLS)_PATH = $(PLATFORM_PATH)/cavm_sdk
-XPSHELL = xpshell.deb
-$(XPSHELL)_PATH = $(PLATFORM_PATH)/cavm_sdk
+include $(PLATFORM_PATH)/cavm-sai.mk
+include $(PLATFORM_PATH)/docker-syncd-cavm.mk
+include $(PLATFORM_PATH)/docker-orchagent-cavm.mk
 
-SONIC_COPY_DEBS += $(CAVM_LIBSAI) $(CAVM_SAI) $(XP_TOOLS) $(XPSHELL)
+SONIC_ALL += $(DOCKER_SYNCD_CAVM) \
+	     $(DOCKER_ORCHAGENT_CAVM)
```
Comment (marian-pritsak) [inline platform/cavium/rules.mk]: You'll also need [this line](https://github.com/Azure/sonic-buildimage/blob/master/platform/broadcom/rules.mk#L1) for building common dockers

Comment (garyachy) [inline platform/cavium/rules.mk]: Hmm, why do I need to replicate "$(DOCKER_SYNCD_CAVM)_DEPENDS" here?

Comment (marian-pritsak) [inline platform/cavium/rules.mk]: You need this one
`include $(PLATFORM_GENERIC_PATH)/rules.mk`
see https://github.com/Azure/sonic-buildimage/commit/e3b126893ba37f0fe3292f9f953c4d9a63d4f631

Comment (marian-pritsak) [inline platform/cavium/rules.mk]: Ooh
Its already there
My fault, sorry

Comment (garyachy) [inline platform/cavium/rules.mk]: Thanks for your review

Comment (marian-pritsak) [inline platform/cavium/rules.mk]: This line was added in last commit
You just have to merge changes introduced in https://github.com/Azure/sonic-buildimage/commit/e3b126893ba37f0fe3292f9f953c4d9a63d4f631

Comment (garyachy) [inline platform/cavium/rules.mk]: Done

**Comment by garyachy** [comment]
@lguohan pls. merge

File: platform/cavium/docker-syncd-cavm/Dockerfile.template [hunk 3]
```diff
@@ -0,0 +1,18 @@
+FROM docker-base
+
+RUN apt-get update
+
+COPY deps /deps
+
+RUN apt-get -y install libpcap-dev libxml2-dev python-dev swig libsensors4-dev
+
+SED_DPKG
```
Comment (lguohan) [inline platform/cavium/docker-syncd-cavm/Dockerfile.template]: SED_DPKG is superceded by jinj2 template in commit [66aebb3](https://github.com/Azure/sonic-buildimage/commit/66aebb329ccec74e92ad25875ea3e8cc2de65055)

**Pull request merged**